src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -2003,14 +2017,22 @@ void SetTotalDifficultyDeep(BlockHeader current)
             }
             else
             {
-                BlockHeader parentHeader = GetParentHeader(header);
-
-                if (parentHeader.TotalDifficulty is null)
+                UInt256 parentTotalDifficulty;
+                if (_syncConfig.FastSync && header.Number - 1 == _syncConfig.PivotNumberParsed)
                 {
-                    SetTotalDifficultyDeep(parentHeader);
+                    parentTotalDifficulty = _syncConfig.PivotTotalDifficultyParsed;
                 }
+                else
+                {
+                    BlockHeader parentHeader = GetParentHeader(header);
```

Comment:
Let's discuss it :) I think we fixed it on master now and we can remove these lines